File: action/src/main/java/move/action/ScheduledActionProvider.java
package move.action;

import javax.inject.Inject;

/**
 *
 */
public class ScheduledActionProvider<A extends Action<Object, Object>> extends ActionProvider<A, Object, Object> {
    private ScheduledAction scheduledAction;

    @Inject
    public ScheduledActionProvider() {
    }

    public ScheduledAction getScheduledAction() {
        return scheduledAction;
    }

    @Override
    protected void init() {
        scheduledAction = getActionClass().getAnnotation(ScheduledAction.class);
        super.init();
    }
}

File: action/src/main/java/move/action/ScheduledActionProvider.kt
package move.action

import io.vertx.rxjava.core.Vertx
import javax.inject.Inject
import javax.inject.Provider

/**

 */
open class ScheduledActionProvider<A : Action<Unit, Unit>> @Inject
constructor(vertx: Vertx,
            actionProvider: Provider<A>)
   : ActionProvider<A, Unit, Unit>(
   vertx,
   actionProvider
) {
   override val isScheduled = true
   val annotation: ScheduledAction = actionClass.getAnnotation(ScheduledAction::class.java)

   override val annotationTimeout: Int
      get() = 0
}
